### DISCUSSION TITLE
Updating distro created by sbt dist

### DISCUSSION THREAD

**Discussion by hmarzooq** [playframework/playframework#11643] [General]
Hi.

We're using `sbt dist` in build stage of CI/CD. The build stage is bit complex and do several other things along with `sbt dist`. In some case it becomes expensive to run build stage when there is just a javascript file update in the distro created by `sbt dist`.

Having said that, we've some javascripts under `/webapp/public/javascripts`. The `sbt dist` creates a zip file `<project-name>-<project-version>.zip ` at `/webapp/target/universal/`. When unzipping `/webapp/target/universal/<project-name>-<project-version>.zip` there is a project jar at `/webapp/target/universal/<project-name>-<project-version>/lib/<project-name>-<project-version>.jar`.  Unzipping this jar includes a javascript folder at: `/webapp/target/universal/<project-name>-<project-version>/lib/ETA-INF/resources/webjars/<project-name>/<project-version>/javascripts`

I want to `replace` a javascript file at `/webapp/target/universal/<project-name>-<project-version>/lib/ETA-INF/resources/webjars/<project-name>/<project-version>/javascripts`. I can do this by unzipping `<project-name>-<project-version>.zip` and then unzipping `<project-name>-<project-version>.jar` and replacing javascript file and then jar the folder and zip it again. But I see there are two more files with same name as of a javascript file with `md5` and `gz` extension. I'm not sure what is their purpose.  And what approach shall I take here to update/replace a javascript file? 

Any guidance is highly appreciated.

**Comment by mkurz** [comment]
The `gz` is an archive containing the compressed javascript file, compressed by the gzip algorithm. So when a client (browser) sends the http header `Accept-Encoding: gzip, ...` then Play will send the `gz` file to the client instead of the plain javascript file (with the response header `Content-Encoding: gzip`).
The `md5` file contains the checksum of the file.

So what you need to do is to also replace both files based on the content of the modfied new js file.
On macOS and Linux you should be able to run the command `md5sum <file.js>` to generate a new checksum value and then use that as content for the md5 file.
To gzip the modified javascript file you can use the `gzip` command, e.g. `gzip filejs`. Then also replace the old gz file with the new one you generated.
